db/db_impl_secondary.cc

```diff
@@ -184,6 +249,37 @@ Status DBImplSecondary::RecoverLogFiles(
         continue;
       }
       WriteBatchInternal::SetContents(&batch, record);
+      std::vector<uint32_t> column_family_ids;
+      status = CollectColumnFamilyIdsFromWriteBatch(batch, &column_family_ids);
+      if (status.ok()) {
+        SequenceNumber seq = versions_->LastSequence();
+        for (const auto id : column_family_ids) {
+          ColumnFamilyData* cfd =
+              versions_->GetColumnFamilySet()->GetColumnFamily(id);
+          if (cfd == nullptr) {
+            continue;
+          }
+          if (cfds_changed->count(cfd) == 0) {
+            cfds_changed->insert(cfd);
+          }
+          auto curr_log_num = std::numeric_limits<uint64_t>::max();
+          if (current_log_.count(cfd) > 0) {
+            curr_log_num = current_log_[cfd];
+          }
+          if (!cfd->mem()->IsEmpty() &&
+              (curr_log_num == std::numeric_limits<uint64_t>::max() ||
+               curr_log_num != log_number)) {
+            const MutableCFOptions mutable_cf_options =
+                *cfd->GetLatestMutableCFOptions();
```

Comment:
just to confirm, we are protected by a mutex here, right? I know `Recover` is but not too sure about `TryCatchUpWithPrimary`

Reply:
Which part are you referring to?

Reply:
Chatted offline. Resolved